spot_ros2_control/hardware/src/spot_hardware_interface.cpp

```diff
@@ -189,14 +198,16 @@ std::vector<hardware_interface::CommandInterface> SpotHardware::export_command_i
 }
 
 hardware_interface::CallbackReturn SpotHardware::on_deactivate(const rclcpp_lifecycle::State& /*previous_state*/) {
-  // Once command streaming is enabled, this should release the lease and stop command streaming
+  release_lease();
   return hardware_interface::CallbackReturn::SUCCESS;
 }
 
 hardware_interface::CallbackReturn SpotHardware::on_shutdown(const rclcpp_lifecycle::State& /*previous_state*/) {
   stop_state_stream();
-  // Once command streaming is enabled, this should also release the lease and stop command streaming
-  // (if not already deactivated)
+  release_lease();
+  if (!power_off()) {
+    return hardware_interface::CallbackReturn::ERROR;
+  }
```

Comment:
@tcappellari-bdai note that we may shutdown from any state. Are all these functions idempotent? Is it safe to release an already released lease. Same for power management, stream management, and so on.